### PR TITLE
ol - improve usage summary support for pg&e

### PR DESCRIPTION
**Pull request by orlandoWexus**
Now, usage summary supports cost_additional_detail_last_period, tariff and read_cycle.

### CODE REVIEW

File: lib/green-button-data/parser/cost_additional_detail_last_period.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      # Special case for PG&E generic namespacing
+      element :'ns0:powerOfTenMultiplier', class: Integer,
+              as: :power_of_ten_multiplier
+      element :'ns0:uom', class: Integer, as: :uom
+      element :'ns0:value', class: Integer, as: :value
+      element :'ns0:note', as: :note
+      element :'ns0:itemKind', class: Integer, as: :item_kind
+    end
+  end
+end
```
Comment (AndrewJo) [inline lib/green-button-data/parser/cost_additional_detail_last_period.rb]: Please add a newline at the end of file.

File: lib/green-button-data/parser/usage_summary.rb [hunk 2]
```diff
@@ -10,8 +10,8 @@ class UsageSummary
       element :billToDate, class: Integer, as: :bill_to_date
       element :costAdditionalLastPeriod, class: Integer,
               as: :cost_additional_last_period
-      element :costAdditionalDetailLastPeriod, class: Integer,
-              as: :cost_additional_detail_last_period
+      elements :costAdditionalDetailLastPeriod, class: CostAdditionalDetailLastPeriod,
```
Comment (AndrewJo) [inline lib/green-button-data/parser/usage_summary.rb]: Please keep the line below 80 characters.

File: lib/green-button-data/parser/usage_summary.rb [hunk 3]
```diff
@@ -59,8 +61,8 @@ def status_time_stamp
       element :'espi:billToDate', class: Integer, as: :bill_to_date
       element :'espi:costAdditionalLastPeriod', class: Integer,
               as: :cost_additional_last_period
-      element :'espi:costAdditionalDetailLastPeriod', class: Integer,
-              as: :cost_additional_detail_last_period
+      elements :'espi:costAdditionalDetailLastPeriod', class: CostAdditionalDetailLastPeriod,
```
Comment (AndrewJo) [inline lib/green-button-data/parser/usage_summary.rb]: Please keep the line below 80 characters.

File: lib/green-button-data/parser/usage_summary.rb [hunk 4]
```diff
@@ -96,8 +100,8 @@ def status_time_stamp
       element :'ns0:billToDate', class: Integer, as: :bill_to_date
       element :'ns0:costAdditionalLastPeriod', class: Integer,
               as: :cost_additional_last_period
-      element :'ns0:costAdditionalDetailLastPeriod', class: Integer,
-              as: :cost_additional_detail_last_period
+      elements :'ns0:costAdditionalDetailLastPeriod', class: CostAdditionalDetailLastPeriod,
```
Comment (AndrewJo) [inline lib/green-button-data/parser/usage_summary.rb]: Please keep the line below 80 characters.

**Comment by AndrewJo** [comment]
@orlandoWexus Looks great! Besides the code styles, it's ready to be merged. Please let me know when you make these modifications and I will merge it in ASAP.

**Comment by orlandoWexus** [comment]
@AndrewJo Thanks for being so responsive. And it's ready!

**Comment by AndrewJo** [comment]
No problem! Merged.
Will release a new minor version update in few minutes.

**Pull request merged**